authorship/data.py

```diff
@@ -60,25 +60,24 @@ def store_data(path, speakers):
         f.write(json.dumps(speakers))
 
 
-all_words = 0
-
 def read_session(lines):
-    global all_words
-    global test
     speakers = []
     test = lines.read()
     test = re.sub('[0-9]*\.[0-9]*\t', '', test)
-    test = re.sub('[A-Za-z]*\*n','',test)
-    test = re.sub('[A-Za-z]*\*u','',test)
-    test = re.sub('[A-Za-z]*\*a','',test)
-    test = re.sub('[A-Za-z]*\*x','',test)
-    test = test.replace('start\tend\ttext\n', '').replace('.', '').replace('-', ' ').replace('?', '').replace('\n',' ').replace('xxx', '').replace('ggg', '').replace('vvv', '').replace('*v','').replace('*s','')
-    s = test
-    s = s.translate({ord(c): None for c in string.punctuation})
-    all_words += len(word_tokenize(s))
-    # print(word_tokenize(s))
-    speakers.extend(word_tokenize(s))
-    #print('all words in session:', all_words)
+
+    # remember:
+    # *v: non-Dutch  words,  *n:  new  non-existing  words,  *s:  street  words,
+    # *a: incomplete  words, *u:  distorted  words, *x: unclear word
+    # we keep the words with their notation (for *n, *a, *u, and *x) to be able to exclude them when we
+    # derive the set of the most frequent words
+    test = test.replace('start\tend\ttext\n', '').replace('.', '').replace('-', ' ').replace('?', '').replace('\n', ' ') \
+        .replace('xxx', '').replace('ggg', '').replace('vvv', '').replace('*v','').replace('*s','')
+    s = test.translate({ord(c): None for c in string.punctuation if c != '*'})
+    tk = WhitespaceTokenizer()
+    conv = tk.tokenize(s)
+    if len(conv) > 30:
```

Comment:
this is a constraint on the input of this function but not visible to a user. I think this should be either a parameter (preferred) or documented

Reply:
thanks for it. i removed it (it was a temp thing to check sth.)